tests/validation/test_validator.py

```diff
@@ -67,23 +79,136 @@ def test_validator_iteration():
 
 
 def test_validator():
```

Comment:
Great! I would suggest one more test, which is when the splitters provides a list of indexes for the testing index, i.e., fold = [train_index, list of test_index]